### PR TITLE
Added brushing functionality

### LINKED ISSUE #33: Brush Filters
Be able to brush over a range and select the data, converting it into a Vega `transform` filter.

### PR DESCRIPTION
**Pull request by Waidhoferj**
https://user-images.githubusercontent.com/33008436/126231777-5f167c75-f873-4f4f-8b22-7e006e01611e.mov

**Brush filtering on the graph!**
Fixes #33 
## Changes
- Brushing listener is not debounced to prevent overly active signal update (adaptable).
- `VegaPandasTranslator` now supports multiple types of compound filter queries: `or`, `and`, and `not`.
- Category filtering screen checkboxes now initialize to the state of the spec (correctly checked or unchecked)
- Updated `updateSpecFilter` to handle multiple compound operators.

## Testing
- Open a scatterplot.
- Drag over a region to create a bounding box.
- Click "Apply changes"
- Try to apply other filters on the filter screens and check that nothing breaks

- Open a bar chart.
- Repeat previous tests.

### Notes
- We could definitely combine the brush with the axis based sliders to make it easier to adjust selections precisely.

### CODE REVIEW

**Comment by jahn96** [comment]
### A few issues with implementation
1. I think brushing the bar chart is not working.
Kapture 2021-07-20 at 11 07 25
2. Exported code and graph seem matched. But, the code and a slider in the filter don't.
Screen Shot 2021-07-20 at 11 13 56 AM

### General thoughts
1. Would it be possible to zoom back out?
2. Thoughts on automatically updating the graph without clicking `apply changes`?

File: src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -112,7 +114,8 @@ const actionBarCss = css`
 
 function ActionBar() {
   const [opHistory, setOpHistory] = useModelState('spec_history');
-  const spec = useModelState('graph_spec')[0];
+  const [spec, setSpec] = useModelState('graph_spec');
+  const [selectedData] = useModelState('selected_data');
```
Comment (jahn96) [inline src/components/Sidebar/Sidebar.tsx:118]: Can we do this if we just want the first element?

Comment (Waidhoferj) [reply src/components/Sidebar/Sidebar.tsx:118]: Sorry I don't know what you are asking here

**Review by jahn96 — COMMENTED** [review]
I left some comments above.

File: src/components/Sidebar/Sidebar.tsx (hunk 1 above)
Comment (jahn96) [reply src/components/Sidebar/Sidebar.tsx:118]: `const [selectedData] = useModelState('selected_data');` useModelState returns an array with two elements. In this case, selectedData would have the first element right?

**Comment by Waidhoferj** [comment]
Thanks this is super helpful, I'll go another round on the implementation.

File: src/components/Sidebar/Sidebar.tsx (hunk 1 above)
Comment (jahn96) [reply src/components/Sidebar/Sidebar.tsx:118]: NVM

Comment (Waidhoferj) [reply src/components/Sidebar/Sidebar.tsx:118]: Yeah it's roughly equivalent to 'const selectedData = useModelState('selected_data')[0]'

**Review by jahn96 — APPROVED** [review]
Looks fantastic! Thanks!

**Pull request merged**